src/server/game/Handlers/NPCHandler.cpp

```diff
@@ -391,12 +391,12 @@ void WorldSession::SendSpiritResurrect()
         if (corpseGrave != ghostGrave)
             _player->TeleportTo(corpseGrave->map_id, corpseGrave->x, corpseGrave->y, corpseGrave->z, _player->GetOrientation());
         // or update at original position
-        else
-            _player->UpdateObjectVisibility();
+        //else
+        //    _player->UpdateObjectVisibility(); called in ResurrectPlayer
```

Comment:
If you are sure 100% that this is not needed, just erase it.
If not, trace the code and debug if necessary till you are 100% sure that this is needed or not.